Code/GraphMol/AdjustQuery.cpp

```diff
@@ -61,6 +61,216 @@ unsigned int parseWhichString(const std::string &txt) {
   }
   return res;
 }
+
+void adjustConjugatedFiveRings(RWMol &mol) {
+  /*
+   The idea here is to allow conjugated five-rings to match either aromatic or
+   aliphatic rings
+
+   five-rings which contain at least 3 conjugated bonds have all of their
+   non-query bonds replaced with a SINGLE|DOUBLE|AROMATIC query.
+
+   */
+
+  std::vector<Bond::BondType> bondTypesToModify = {
+      Bond::BondType::SINGLE, Bond::BondType::DOUBLE, Bond::BondType::AROMATIC};
+  if (!mol.getRingInfo()->isInitialized()) {
+    MolOps::symmetrizeSSSR(mol);
+  }
+  for (auto ring : mol.getRingInfo()->bondRings()) {
+    // only consider 5-rings with at least 3 conjugated bonds
+    if (ring.size() != 5) {
+      continue;
+    }
+    unsigned int nconj = 0;
+    for (auto bi : ring) {
+      const auto bond = mol.getBondWithIdx(bi);
+      if (bond->getIsConjugated()) {
+        ++nconj;
+        if (nconj >= 3) {
+          break;
+        }
+      }
+    }
+    if (nconj < 3) {
+      continue;
+    }
+    // now make the adjustments
+    // FIX: replace this with the SingleOrDoubleOrAromatic query once the
+    // tautomer PR is merged
+    QueryBond qb;
+    qb.setQuery(makeBondOrderEqualsQuery(Bond::BondType::SINGLE));
+    qb.expandQuery(makeBondOrderEqualsQuery(Bond::BondType::DOUBLE),
+                   Queries::COMPOSITE_OR);
+    qb.expandQuery(makeBondOrderEqualsQuery(Bond::BondType::AROMATIC),
+                   Queries::COMPOSITE_OR);
+    for (auto bi : ring) {
+      const auto bond = mol.getBondWithIdx(bi);
+      if (!bond->hasQuery() &&
```

Comment:
I thought about that, but it seemed like that's more likely to just be confusing.

Reply:
Well, my point for the "AND" query is that, in case of a clash, it would return an empty result (which would be right, since both clashing conditions cannot be satisfied at the same time), rather than a wrong one which would satisfy just one of the clashing conditions.

But logging a message in case of clashing conditions would also be acceptable.

Reply:
As I was writing the code to log a message I came up with an even better solution:
if the bond already has a query, replace the query but include a log message that this was done.
That way the output of the function matches what it says it will do

Reply:
I think that makes sense.